### PR TITLE
add llama model type

### PR DESCRIPTION
**Pull request by Lazarus-931**
Refractor to move llama models from mlx-lm's dependency to full mlx-vlm native support, moving it to models/llama/. 

Llama / Llama-3.x text-only checkpoints now load and run natively through mlx-vlm instead of routing through the text_only.py mlx-lm fallback.

I verified it a/b on checkpoints, with a test_llama_language_model in test_models.py that covers config, sanitize (idempotency + language_model. prefixing), cache round-trip, and finite logits.

All tests green so far

### CODE REVIEW

File: mlx_vlm/tests/test_utils.py [hunk 1]
```diff
@@ -523,7 +523,10 @@ def __call__(self, inputs, cache=None):
             return self.model(inputs)
 
     with (
-        patch("mlx_vlm.utils.load_config", return_value={"model_type": "llama"}),
+        patch(
+            "mlx_vlm.utils.load_config",
+            return_value={"model_type": "unvendored_text_arch"},
+        ),
```
Comment (Blaizzy) [inline mlx_vlm/tests/test_utils.py:529]: I don't get this change?

**Review by Blaizzy — APPROVED** [review]
LGTM, just one nit

File: mlx_vlm/tests/test_utils.py (hunk 1 above)
Comment (Lazarus-931) [reply mlx_vlm/tests/test_utils.py:529]: tests atm check mlx-lm fallback for "llama", swapping it for fake model type so the tests keep testing the fallback no matter which models we vendor next

Comment (Blaizzy) [reply mlx_vlm/tests/test_utils.py:529]: Ok, make sense!

**Comment by Blaizzy** [comment]
Please resolve the conflicts 👌🏽

**Comment by Lazarus-931** [comment]
Conflicts resolved — merged latest main (the new qwen3 model type test landed next to the llama one in test_models.py, kept both). Tests pass.

**Review by Blaizzy — APPROVED** [review]
